reascripts/ReaSpeech/source/ReaSpeechWorker.lua

```diff
@@ -198,9 +185,24 @@ function ReaSpeechWorker:handle_job_status(active_job, response)
   end
 
   if response.job_status == 'SUCCESS' then
-    local transcript_url_path = response.job_result.url_path
+    local job_result = response.job_result
+    if job_result.result then
+      self:handle_response(active_job, job_result.result)
+      self.active_job = nil
+      return false
+    end
+
     response._job = active_job.job
-    active_job.transcript_request = ReaSpeechAPI:fetch_large(transcript_url_path)
+
+    if response.job_result.url_path then
+      table.insert(active_job.requests, ReaSpeechAPI:fetch_large(response.job_result.url_path))
+    elseif response.job_result.url_paths then
+      for _, url_path in pairs(response.job_result.url_paths) do
+        table.insert(active_job.requests, function()
+           ReaSpeechAPI:fetch_large(url_path)
+        end)
+      end
+    end
```

Comment:
we're looking for either a single `url_path` or a collection `url_paths`, but in both cases are treating the followup request handling as a sequence of 1 or n.

in the `n` case, the requests are wrapped in a function to prevent them firing off right away.